files/file.py

```diff
@@ -428,6 +432,47 @@ def main():
 
         module.exit_json(dest=path, changed=True, diff=diff)
 
+    elif state == "empty":
+        # NOTE(robputt796): Initial response to https://github.com/ansible/ansible-modules-core/issues/902
+        # Allows truncate of file, or removal of sub files for a directory for empty state.
+        if not module.check_mode:
+            # Empty state can only be used against a file (to truncate) or directory (to empty)
+            if prev_state in ['file', 'directory']:
+                changed = False
+                try:
+                    if prev_state == 'file':
+                        if os.stat(path).st_size != 0:
+                            # File not empty, truncate.
+                            try:
+                                file = open(path, 'w')
+                                file.truncate(0)
+                                changed = True
+                                file.close()
+                            except IOError, e:
+                                module.fail_json(path=path,
+                                                 msg='There was an issue ensuring %s is empty: %s' % (path, str(e)))
+
+                    if prev_state == 'directory':
+                        file_listing = os.listdir(path)
+                        if file_listing != []:
+                            # Directory is not empty, emptying
+                            changed = True
+                            for file_name in file_listing:
+                                full_path = "%s/%s" % (path, file_name)
+                                if os.path.isdir(full_path):
+                                    shutil.rmtree(full_path)
```

Comment:
`
$ mkdir foo
$ ln -s foo bar
$ python
>>> import os.path
>>> os.path.isdir('bar')
True
>>> import shutil
>>> shutil.rmtree('bar')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/shutil.py", line 232, in rmtree
    onerror(os.path.islink, path, sys.exc_info())
  File "/usr/lib/python2.7/shutil.py", line 230, in rmtree
    raise OSError("Cannot call rmtree on a symbolic link")
OSError: Cannot call rmtree on a symbolic link
`